channeldb/channel_test.go

```diff
@@ -978,6 +978,91 @@ func TestChannelStateTransition(t *testing.T) {
 	require.Empty(t, fwdPkgs, "no forwarding packages should exist")
 }
 
+// TestOpeningChannelTxConfirmation verifies that calling MarkConfirmationHeight
+// correctly updates the confirmed state. It also ensures that calling Refresh
+// on a different OpenChannel updates its in-memory state to reflect the prior
+// MarkConfirmationHeight call.
+func TestOpeningChannelTxConfirmation(t *testing.T) {
+	t.Parallel()
+
+	fullDB, err := MakeTestDB(t)
+	require.NoError(t, err, "unable to make test database")
+
+	cdb := fullDB.ChannelStateDB()
+
+	// Create a pending channel that was broadcast at height 99.
+	const broadcastHeight = uint32(99)
+	channelState := createTestChannel(
+		t, cdb, pendingHeightOption(broadcastHeight),
+	)
+
+	// Fetch pending channels from the database.
+	pendingChannels, err := cdb.FetchPendingChannels()
+	require.NoError(t, err, "unable to list pending channels")
+	require.Len(t, pendingChannels, 1, "expected one pending channel")
+
+	// Verify the broadcast height of the pending channel.
+	require.Equal(
+		t, broadcastHeight, pendingChannels[0].FundingBroadcastHeight,
+		"broadcast height mismatch",
+	)
+
+	confirmationHeight := broadcastHeight + 1
+
+	// Mark the channel's confirmation height.
+	err = pendingChannels[0].MarkConfirmationHeight(confirmationHeight)
+	require.NoError(t, err, "unable to mark channel's confirmation height")
+
+	// Ensure the channel remains pending after confirmation.
+	require.True(t, pendingChannels[0].IsPending, "channel should remain "+
```

Comment:
Q: Not sure if this case here makes sense, because the confirmation height is not related to the pending status directly.